src/towncrier/test/test_builder.py

```diff
@@ -132,3 +134,59 @@ def test_orphan_all_digits(self):
             parse_newfragment_basename("+123.feature", ["feature"]),
             ("+123", "feature", 0),
         )
+
+
+class TestIssueOrdering(TestCase):
+    template = dedent(
+        """
+    {% for section_name, category in sections.items() %}
+    {% if section_name %}# {{ section_name }}{% endif %}
+    {%- for category_name, issues in category.items() %}
+    ## {{ category_name }}
+    {% for issue, numbers in issues.items() %}
+    - {{ issue }}{% if numbers %} ({{ numbers|join(', ') }}){% endif %}
+
+    {% endfor %}
+    {% endfor -%}
+    {% endfor -%}
+    """
+    )
+
+    def render(self, fragments):
+        return render_fragments(
+            template=self.template,
+            issue_format=None,
+            fragments=fragments,
+            definitions={},
+            underlines=[],
+            wrap=False,
+            versiondata={},
+        )
+
+    def test_ordering(self):
+        """
+        Issues are ordered by their number, not lexicographically.
```

Comment:
Not sure if this is correct.

I see gh-2 is ordered before gh-1.

I guess the docstring is missing some details.


```suggestion
        When a single fragment is associated with multiple issue numbers, the issue references are sorted by their number, not lexicographically.
```

Reply:
Again, this primarily is a fix to overall ordering, not just when a fragment has multiple numbers. The primary sort key is the non-numeric portion and specifically with non-numbers first for backwards compatibility, hence why (random, gh-2) occurs before (gh-1)